src/components/partner/partner.service.ts

```diff
@@ -119,31 +123,68 @@ export class PartnerService {
       };
     }
 
-    const changes = this.repo.getActualChanges(object, input);
-    this.privileges.for(session, Partner, object).verifyChanges(changes);
+    const changes = this.repo.getActualChanges(partner, input);
+    this.privileges.for(session, Partner, partner).verifyChanges(changes);
     const {
       pointOfContactId,
       languageOfWiderCommunicationId,
+      fieldRegions,
       ...simpleChanges
     } = changes;
 
-    await this.repo.updateProperties(object, simpleChanges);
+    await this.repo.updateProperties(partner, simpleChanges);
 
     if (pointOfContactId) {
       await this.repo.updatePointOfContact(input.id, pointOfContactId, session);
     }
+
     if (languageOfWiderCommunicationId) {
       await this.repo.updateRelation(
         'languageOfWiderCommunication',
         'Language',
-        object.id,
+        partner.id,
         languageOfWiderCommunicationId,
       );
     }
 
+    if (fieldRegions) {
+      await this.validateFieldRegionIds(
+        fieldRegions,
+        'FieldRegion',
+        'fieldRegionId',
+        'Field region not found',
+      );
+
+      await this.repo.updateRelationList({
+        id: partner.id,
+        relation: 'fieldRegions',
+        newList: fieldRegions,
+      });
+    }
+
     return await this.readOne(input.id, session);
   }
 
+  protected async validateFieldRegionIds(
```

Comment:
Let's change this to the new `getBaseNodes` query. So `O(N) -> O(1)`.
Also we can defer this to after the update query and only run if the expected length is different from the actual length, and then determine which IDs are invalid.